src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -234,24 +267,100 @@ public override Task<HttpResponseMessage> InvokeMethodWithResponseAsync(HttpRequ
 
             // Note: we intentionally DO NOT validate the status code here.
             // This method allows you to 'invoke' without exceptions on non-2xx.
-            return this.httpClient.SendAsync(request, cancellationToken);
+            try
+            {
+                return await this.httpClient.SendAsync(request, cancellationToken);
+            }
+            catch (HttpRequestException ex)
+            {
+                // Our code path for creating requests places these keys in the request properties. We don't want to fail
+                // if they are not present.
+                request.Properties.TryGetValue(AppIdKey, out var appId);
+                request.Properties.TryGetValue(MethodNameKey, out var methodName);
+
+                throw new InvocationException(
```

Comment:
question - why do we have a special InvocationException class here whereas everything else uses DaprException?

Reply:
This is a subclass of DaprException - so a catch for DapException would handle this as well. The reason for the subclass is that we have more information and context to include for invocation. Also needing to handle errors in this case is *much* more common since invoke calls business logic on the side - error cases are **expected**.

Basically, it makes sense to try and optimize this experience. The errors in the other cases are fairly generic, and could be improved in the future as needed based on feedback (without breaking changes).

Reply:
ok...makes sense